src/rules/no-named-as-default-member.js

```diff
@@ -49,6 +49,8 @@ module.exports = function(context) {
   }
 
   function handleDestructuringAssignment(node) {
+    if (!node.init) return
+
```

Comment:
> I generally like to do lots of simple predicates vs. mega-predicates

When there are too many conditions, I completely agree. Another way that I like, is to create functions and call them in one condition.

```js
function isDestructure(node) {
  return node.id.type === 'ObjectPattern' && node.init.type === 'Identifier';
}

if (!node.init || !isDestructure(node)) {
  return;
}
```